src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,475 @@
+        40 HYYYYZ, 41 HYYYYY, 42 HXZZZZ, 43 HXYZZZ, 44 HXYYZZ,
+        45 HXYYYZ, 46 HXYYYY, 47 HXXZZZ, 48 HXXYZZ, 49 HXXYYZ,
+        50 HXXYYY, 51 HXXXZZ, 52 HXXXYZ, 53 HXXXYY, 54 HXXXXZ, 55 HXXXXY,
+        56 HXXXXX
+        """
+
+        prim_types = []
+        for atom in self.ccdata.gbasis:
+            for prims in atom:
+                prim_orb = []
+                for i in range(self.num_orb[prims[0]]):
+                    prim_orb += [(self.idx_orb[prims[0]]  + i)]\
+                                * len(prims[1])
+                prim_types += prim_orb
+        return prim_types
+
+    def _prim_types_section(self):
+        prim_types = self._prim_types()
+        # GAMESS specific reordering.
+        if self.ccdata.metadata['package'] == 'GAMESS':
+            prim_types = self._rearrange_modata(prim_types)
+        return self._list_format(prim_types, 10, '%d ')
+
+    def _prim_exps(self):
+        """Space-separated list of primitive exponents."""
+        prim_exps = []
+        for atom in self.ccdata.gbasis:
+            for prims in atom:
+                prim_exps += [prim[0] for prim in prims[1]]\
+                            * self.num_orb[prims[0]]
+        return self._list_format(prim_exps, 5)
+
+    def _mo_occup_nos(self):
+        """MO occupation numbers."""
+        occup = []
+        electrons = self._no_electrons()
+        alpha = self._no_alpha_electrons()
+        beta = self._no_beta_electrons()
+        if len(self.ccdata.homos) == 1:
+            occup += ['{:22.11E}'.format(2)] * int(electrons / 2) +\
+                        ['{:22.11E}'.format(1)] * (electrons % 2)
+        else:
+            occup += ['{:22.11E}'.format(1)] *  + alpha +\
+                        ['{:22.11E}'.format(1)] * beta
+        return occup
+
+    def _mo_energies(self):
+        """List of MO Energies."""
+        mo_energies = []
+        alpha_elctrons = self._no_alpha_electrons()
+        beta_electrons = self._no_beta_electrons()
+        for mo_energy in self.ccdata.moenergies[0][:alpha_elctrons]:
+            mo_energies.append('{:22.11E}'.format(
+                utils.convertor(mo_energy, 'eV', 'hartree')))
+        if self.ccdata.mult > 1:
+            for mo_energy in self.ccdata.moenergies[1][:beta_electrons]:
+                mo_energies.append('{:22.11E}'.format(
+                    utils.convertor(mo_energy, 'eV', 'hartree')))
+        return mo_energies
+
+    def _mo_spin_types(self):
+        """MO Spin Types."""
+        spin_types = []
+        electrons = self._no_electrons()
+        alpha = self._no_alpha_electrons()
+        beta = self._no_beta_electrons()
+        if len(self.ccdata.homos) == 1:
+            spin_types += ['Alpha and Beta'] * int(electrons / 2) +\
+                            ['Alpha'] * (electrons % 2)
+        else:
+            spin_types += ['Alpha'] * alpha +\
+                            ['Beta'] * beta
+        return spin_types
+
+    pi3 = (2.0 / numpy.pi) ** 3
```

Comment:
`normalize` is called very frequently, once for each `mocoeff` .Hence, I had put it as a class attribute to avoid recalculation. How about making it a global variable?